### PR TITLE
 refactor AmmImpl.create() to use only 2 arguments: provider, poolAddress

### PR DESCRIPTION
**Pull request by satoshinakamotonetwork**
-  refactor `AmmImpl.create()` to use only 2 arguments: provider, poolAddress
- use new `AmmImpl.create()` function in examples: swap, swap_quote, get_pool_info
- update `CHANGELOG.md`
- update vault-sdk to 2.0.0
- refactor types and functions to use tokenMint instead of tokenInfo
- provide address and decimals in 2 object: `tokenAMint` and `tokenBMint`
- differentiate `tokenMint` and `tokenAddress(`PublickKey Only)

### CODE REVIEW

File: ts-client/src/amm/index.ts [hunk 1]
```diff
@@ -796,6 +794,13 @@ export default class AmmImpl implements AmmImplementation {
...
     const poolState = await getPoolState(pool, ammProgram);
+    const { tokenAMint, tokenBMint } = poolState
+    const [tokenASupply, tokenBSupply] = await Promise.all([
+      provider.connection.getTokenSupply(poolState.tokenAMint),
+      provider.connection.getTokenSupply(poolState.tokenBMint)
+    ])
+    const tokenInfoA = { address: tokenAMint.toString(), decimals: tokenASupply.value.decimals }
+    const tokenInfoB = { address: tokenBMint.toString(), decimals: tokenBSupply.value.decimals }
 
     invariant(tokenInfoA.address === poolState.tokenAMint.toBase58(), `TokenInfoA provided is incorrect`);
```
Comment (andrewsource147) [inline ts-client/src/amm/index.ts:805]: no need to add the check here

**Comment by bingnetwork** [comment]
- update vault-sdk to 2.0.0
- differentiate tokenMint and tokenAddress(PublickKey Only)

File: ts-client/package.json [hunk 2]
```diff
@@ -44,6 +44,9 @@
   "peerDependencies": {
     "@solana/buffer-layout": "^3 || ^4"
   },
+  "resulutions": {
```
Comment (00xSam) [inline ts-client/package.json:47]: typo

File: ts-client/src/amm/index.ts [hunk 3]
```diff
@@ -827,15 +835,17 @@ export default class AmmImpl implements AmmImplementation {
...
-    invariant(tokenInfoB.address === poolState.tokenBMint.toBase58(), `TokenInfoB provided is incorrect`);
-    invariant(tokenInfoA, `TokenInfo ${poolState.tokenAMint.toBase58()} A not found`);
-    invariant(tokenInfoB, `TokenInfo ${poolState.tokenBMint.toBase58()} A not found`);
+    const { tokenAMint, tokenBMint } = poolState
+    const [tokenASupply, tokenBSupply] = await Promise.all([
+      provider.connection.getTokenSupply(tokenAMint),
+      provider.connection.getTokenSupply(tokenBMint)
+    ])
+    const tokenADecimals = tokenASupply.value.decimals
+    const tokenBDecimals = tokenBSupply.value.decimals
```
Comment (00xSam) [inline ts-client/src/amm/index.ts:844]: can we do things similar like `vault-sdk` ? use `getMint` instead and pass the whole `Mint` to AmmImpl instance

**Review by andrewsource147 — APPROVED** [review]

**Pull request merged**